cheroot/_compat.py

```diff
@@ -14,6 +14,14 @@
 except ImportError:
     import selectors2 as selectors  # noqa: F401  # lgtm [py/unused-import]
 
+import socket  # noqa: F401, E501  # lgtm [py/unused-import] lgtm [py/import-and-import-from]
```

Comment:
They just monkey-patch the socket module: https://github.com/mhils/backports.socketpair/blob/master/backports/socketpair/__init__.py#L56

It happens to the globally cached module object, the import doesn't change anything

```suggestion
```